.github/workflows/package-python.yml

```diff
@@ -0,0 +1,80 @@
+name: Create Python Package
+
+#######
+# Creates a Python package via `tox -e package` and uploads it as an artifact named 'packages-<repo name>'.
+#######
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'Python version to use; defaults to latest Python release.'
+        default: '3.X'
+        type: string
+      repository:
+        description: 'GitHub repository to checkout; defaults to repo running this workflow.'
+        default: ${{ github.repository }}
+        type: string
+      tox-source:
+        description: 'The arguments for `pip install` to install tox; use ./path/to/package[dev] for the repo''s pinned version.'
+        default: '.[dev]'
+        type: string
+      build-subdirectory:
+        description: 'The subdirectory to build as a wheel.'
+        default: ''
+        type: string
+      distribution-path:
+        description: 'Relative filepath to distribution(s); e.g. */dist/*'
+        default: 'dist'
+        type: string
+      runner-os:
+        description: 'Runner OS to use to run tox; defaults to ubuntu-latest'
+        default: 'ubuntu-latest'
+        type: string
+    outputs:
+      artifact-name:
+        description: 'Name of the uploaded artifact; use for artifact retrieval.'
+        value: ${{ jobs.package.outputs.artifact-name }}
+
+jobs:
+  package:
+    name: Python Package
+    runs-on: ${{ inputs.runner-os }}
+    outputs:
+      artifact-name: packages-${{ steps.package.outputs.name }}
+    steps:
+
+      - name: Package name
+        id: package
+        run: echo "name=$(basename '${{ inputs.repository }}')" >> ${GITHUB_OUTPUT}
+
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
+        with:
+          repository: ${{ inputs.repository }}
+          fetch-depth: 0  # Fetch all refs so setuptools_scm can generate the correct version number
+
+      - name: Setup Python
+        uses: actions/setup-python@v4.5.0
+        with:
+          python-version: ${{ inputs.python-version }}
+          cache: 'pip'
+          cache-dependency-path: '**/setup.cfg'
```

Comment:
As above - do we want to add `pyproject.toml` and `setup.py` to this list?

Reply:
So....this starts to get a little tricky. The `setup-python` action hard errors if the files specified here don't exist. I'll give it a go and see if all the repos have these files.

(On a related note, this is why some of the other uses of `setup-python` don't/can't use this built in caching.

Reply:
I can save you the trouble; pyproject *should* exist everywhere (and if it doesn't, that's a bug); setup.py currently does, but *should* be slowly disappearing. (I've deleted it in the deb PR branch, for example).

Reply:
Oh - and the template repositories won't have any of these files - but then, we won't be building Python packages for those repos.

Reply:
The tests finished after I added `pyproject.toml` alongside `setup.cfg`.